opentera-webrtc-native-client/OpenteraWebrtcNativeClient/python/package/CMakeLists.txt

```diff
@@ -1,41 +1,118 @@
+endif()
 
+add_custom_command(
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/dist.md5
+    DEPENDS ${HTML_DOC_DEPENDENCY_OPTION}
+    COMMAND /usr/bin/env python3 setup.py bdist bdist_wheel sdist
+    COMMAND tar c ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client/dist 2> /dev/null | md5sum > ${CMAKE_CURRENT_BINARY_DIR}/dist.md5
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client
+    VERBATIM
+)
+add_custom_target(
+    python-opentera-webrtc-native-client-wheel
+    ALL
+    DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/dist.md5
+    VERBATIM
+)
 
-    install(FILES ${LIBRARY_OUTPUT_PATH}/opentera_webrtc_native_client.html DESTINATION python/package)
-endif()
+file(GENERATE
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/cmake/pip_install.cmake
+    CONTENT
+    "execute_process(
+        COMMAND /usr/bin/env python3 -m pip install --upgrade --force-reinstall --no-deps --no-cache-dir --no-index --find-links=${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client/dist opentera-webrtc-native-client
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client
+    )
+    execute_process(
+        COMMAND bash -c \"tar c $(/usr/bin/env python3 -m pip show opentera-webrtc-native-client | perl -n0777E 'say $1 if /Location:\\\\s+(.*?)\\\\n.*/s')/opentera_webrtc_native_client 2> /dev/null | md5sum > ${CMAKE_CURRENT_BINARY_DIR}/install.md5\"
+        WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client
+    )"
+)
 
-# Install files
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/setup.py DESTINATION python/package)
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/README.md DESTINATION python/package)
-install(FILES ${CMAKE_CURRENT_BINARY_DIR}/__init__.py DESTINATION python/package)
-install(FILES ../../../../LICENSE DESTINATION python/package)
-
-
-# Reinstall python library (could be better)
-install(TARGETS opentera_webrtc_native_client DESTINATION python/package)
-
-# Create python wheel package
-add_custom_target(python-wheel-package
-	COMMAND python3 setup.py bdist bdist_wheel sdist
-	WORKING_DIRECTORY ${CMAKE_INSTALL_PREFIX}/python/package)
-	
-# Signaling server files
-set(signaling_server_source_files
-    ../../../../signaling-server/signaling_server.py
-    ../../../../signaling-server/room_manager.py
-    ../../../../signaling-server/requirements.txt
-    ../../../../signaling-server/README.md)
-         
-install(FILES ${signaling_server_source_files} DESTINATION python/signaling-server)
+add_custom_command(
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/install.md5
+    DEPENDS python-opentera-webrtc-native-client-wheel ${CMAKE_CURRENT_BINARY_DIR}/dist.md5
+    COMMAND ${CMAKE_COMMAND} -P ${CMAKE_CURRENT_BINARY_DIR}/cmake/pip_install.cmake
+    WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR}/opentera_webrtc_native_client
+    VERBATIM
+)
+add_custom_target(
+    python-opentera-webrtc-native-client-install
+    ${ALL_OPTION}
+    DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/install.md5
+)
```

Comment:
Install AFTER the wheel is generated, and only if install is set to be done on build (`${ALL_OPTION}` is `ALL` if set to install on build, and empty otherwise)